lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -373,6 +374,112 @@
         version_added: "2.8"
 
+    entrust_cert_type:
+        description:
+            - The type of certificate product to request.
+        type: str
+        default: STANDARD_SSL
+        choices: [ 'STANDARD_SSL', 'ADVANTAGE_SSL', 'UC_SSL', 'EV_SSL', 'WILDCARD_SSL', 'PRIVATE_SSL', 'PD_SSL', 'CODE_SIGNING', 'EV_CODE_SIGNING',
+        'CDS_INDIVIDUAL', 'CDS_GROUP', 'CDS_ENT_LITE', 'CDS_ENT_PRO', 'SMIME_ENT' ]
+        version_added: "2.9"
+
+    entrust_requester_email:
+        description:
+            - The email of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_name:
+        description:
+            - The name of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_phone:
+        description:
+            - The phone number of the requester of the certificate (for tracking purposes).
+        type: str
+        version_added: "2.9"
+
+    entrust_api_user:
+        description:
+            - The username for authentication to the Entrust ECS API.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_key:
+        description:
+            - The key (password) for authentication to the Entrust ECS API.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_client_cert_path:
+        description:
+            - The path of the client certificate used to authenticate to the Entrust ECS API.
+        type: path
+        version_added: "2.9"
+
+    entrust_api_client_cert_key_path:
+        description:
+            - The path of the key for the client certificate used to authenticate to the Entrust ECS API
+        type: path
+        version_added: "2.9"
+
+    entrust_reissue_if_expires_in:
+        description:
+            - If this value is set and an existing certificate is present, it will be reissued if it expires within this number of seconds.
+            - If not present or value is 0, it will be reissued if the certificate is already expired.
+            - For example, if you want to reissue a certificate if it is expiring within a month, set this to 2592000.
+              Certificates with more seconds remainingpast this number of seconds from when the module is called will be left as is.
+        type: int
+        version_added: "2.9"
+
+    entrust_not_after:
+        description:
+            - The point in time at which the certificate stops being valid.
+            - Time can be specified either as relative time or as absolute timestamp.
+            - Time will always be interpreted as UTC.
+            - Valid format is C([+-]timespec | ASN.1 TIME) where timespec can be an integer
+              + C([w | d | h | m | s]) (e.g. C(+32w1d2h).
+            - Note that if using relative time this module is NOT idempotent.
+            - If this value is not specified, the certificate will stop being valid 365 days from now.
```

Comment:
Does the Entrust API allows arbitrary values here? If not, you should mention that there are additional conditions (like must not be too far in the future).

Reply:
Yes, but you're right in that there are limits, though they're largely the 'universal' limits to cert length. I'll document boundaries.

Reply:
I think it's already enough to mention that CA limits apply. Does entrust allow you to freely choose the notAfter timestamp (as long as it satisfies the general limits), or is the provided value taken as a guideline and adjusted in some way? This would also be of interest to a user.

Reply:
It's actually on our roadmap to fix - we currently don't support timestamp, so that's a good catch that it needs to be clearly documented, since providers like selfsigned/ownca do.